path_calc.hpp

```diff
@@ -0,0 +1,83 @@
+#ifndef IRODS_INDEXING_PATH_CALC_HPP
+#define IRODS_INDEXING_PATH_CALC_HPP
+
+#include <set>
+#include <map>
+#include <string>
+#include <functional>
+
+using namespace std::string_literals;
```

Comment:
This shouldn't be in the global namespace. It needs to be moved into the function that uses it.

Reply:
good call.